pkg/providers/aws/provider_blobstorage.go

```diff
@@ -404,6 +405,11 @@ func (p *BlobStorageProvider) reconcileBucketCreate(ctx context.Context, s3svc s
 		}
 	}
 	if foundBucket != nil {
+		if annotations.Has(bs, resourceIdentifierAnnotation) {
+			errMsg := fmt.Sprintf("BlobStorage CR %s in %s namespace has %s annotation with value %s, but no corresponding S3 Bucket was found",
+				bs.Name, bs.Namespace, resourceIdentifierAnnotation, bs.ObjectMeta.Annotations[resourceIdentifierAnnotation])
+			return croType.StatusMessage(errMsg), fmt.Errorf(errMsg)
```

Comment:
Odd thing: this is now being hit on my cluster for some reason, for a bucket that should exist already.

Also, in the status block of the CR, the message seems to be shown 3 times:

```
status:
    message: 'BlobStorage CR example-blobstorage in cloud-resource-operator namespace
      has resourceIdentifier annotation with value gryandvd2dcloudresourceoperatorexam-2dwz,
      but no corresponding S3 Bucket was found: BlobStorage CR example-blobstorage
      in cloud-resource-operator namespace has resourceIdentifier annotation with
      value gryandvd2dcloudresourceoperatorexam-2dwz, but no corresponding S3 Bucket
      was found: BlobStorage CR example-blobstorage in cloud-resource-operator namespace
      has resourceIdentifier annotation with value gryandvd2dcloudresourceoperatorexam-2dwz,
      but no corresponding S3 Bucket was found'
```

Reply:
> Odd thing: this is now being hit on my cluster for some reason, for a bucket that should exist already.

Well I fixed this issue at least. The problem was that I can't read: I had it in the `if foundBucket != nil` block, when I thought that said `if foundBucket == nil`.